src/app/[slug]/page.jsx

```diff
@@ -1,20 +1,16 @@
 import { SinglePage } from '@/queries/pages'
 import { RichText } from '@graphcms/rich-text-react-renderer'
 import { notFound } from 'next/navigation'
+import { draftMode } from 'next/headers'
+import { client } from '@/utils/client'
 
 async function getPage(slug) {
-  const { page } = await fetch(process.env.HYGRAPH_ENDPOINT, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
-      query: SinglePage,
-      variables: { slug: slug }
-    })
-  })
-    .then((res) => res.json())
-    .then((res) => res.data)
+  const { isEnabled } = draftMode()
+  if (isEnabled) client.setHeader('Authorization', `Bearer ${process.env.HYGRAPH_PREVIEW_TOKEN}`)
```

Comment:
I believe we can do this logic inside the `client` file, but we'd have to convert the client into a function

Reply:
Much cleaner that way, for sure! Refactored in latest commit